Lib/subprocess.py

```diff
@@ -1480,7 +1480,26 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
             if shell:
                 startupinfo.dwFlags |= _winapi.STARTF_USESHOWWINDOW
                 startupinfo.wShowWindow = _winapi.SW_HIDE
-                comspec = os.environ.get("COMSPEC", "cmd.exe")
+                # gh-101283: without a fully-qualified path, before Windows
+                # checks the system directories, it first looks in the
+                # application directory, and also the current directory if
+                # NeedCurrentDirectoryForExePathW(ExeName) is true, so try
+                # to avoid executing unqualified "cmd.exe".
+                comspec = os.environ.get('ComSpec')
+                if not comspec:
+                    system_drive = os.environ.get('SystemDrive') or 'C:'
+                    system_root = os.environ.get('SystemRoot') or os.path.join(
+                        system_drive, 'Windows')
+                    comspec = os.path.join(system_root, 'System32', 'cmd.exe')
+                    if not os.path.isfile(comspec):
+                        # cmd.exe is missing, or the system environment
+                        # variables are broken, or they're undefined and the
+                        # system is installed into a non-standard location.
+                        # This is highly unlikely, and we cannot help here.
```

Comment:
Should a RuntimeWarning be raised or logging.warn call made in this case as an FYI?

Reply:
That's a good idea. How about this?

```
                        comspec = 'cmd.exe'
                        warnings.warn(f'spawning "{comspec}" using a '
                                      'relative file path',
                                      RuntimeWarning)
```

Reply:
if we keep the fallback at all rather than just make it an error, something like that.  probably with a stacklevel= set.

related to my other comment, i wouldn't want to add a new warning in a security backport either.

Reply:
@gpshead Thank you for the advice; applied.